src/vellum/lib/test_suite.py

```diff
@@ -0,0 +1,175 @@
+            return
+
+        self._refresh_test_suite_run_state()
+
+        while self.test_suite_run_state in {TestSuiteRunState.QUEUED, TestSuiteRunState.RUNNING}:
+            time.sleep(1)
+            self._refresh_test_suite_run_state()
+            # TODO: Max retries?
+
+        if self.test_suite_run_state == TestSuiteRunState.FAILED:
+            raise Exception("Test suite run failed")
+
+        if self.test_suite_run_state == TestSuiteRunState.CANCELLED:
+            raise Exception("Test suite run was cancelled")
+
+        response = self.client.test_suite_runs.list_executions(
+            self.id, expand=["results.metric_results.metric_definition"]
+        )
+        self.test_suite_run_executions = response.results
+
+    def get_metric_outputs(
+        self, metric_identifier: str | None = None, output_identifier: str | None = None
+    ) -> list[list[TestSuiteRunMetricOutput]]:
+        self._ensure_metric_results_are_loaded()
+
+        all_metric_results = [execution.metric_results for execution in self.test_suite_run_executions]
+
+        filtered_metric_outputs = [
+            metric_result.outputs
+            for metric_result_execution in all_metric_results
+            for metric_result in metric_result_execution
+            if not metric_identifier
+            or (metric_result.metric_id == metric_identifier)
+            or (metric_result.metric_label == metric_identifier)
+            or (metric_result.metric_definition and metric_result.metric_definition.id == metric_identifier)
+            or (metric_result.metric_definition and metric_result.metric_definition.name == metric_identifier)
+            or (metric_result.metric_definition and metric_result.metric_definition.label == metric_identifier)
+        ]
+
+        filtered_metric_filtered_outputs = [
+            [
+                metric_output
+                for metric_output in metric_output_execution
+                if not output_identifier
+                or metric_output.name == output_identifier
+            ]
+            for metric_output_execution in filtered_metric_outputs
+        ]
+
+        return filtered_metric_filtered_outputs
+
+
+ValidOutput = Union[
+    str,
+    float,
+    dict[str, Any],
+    list[SearchResult],
+    list[ChatMessage],
+    VellumError,
+]
+
+
+class VellumTestSuite:
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.client = Vellum(
+            api_key=os.environ.get("VELLUM_API_KEY"),
+        )
+        self.test_suite_id = self._get_test_suite_id_by_name(name)
+
+    def _get_test_suite_id_by_name(self, name: str) -> str:
+        # TODO: We need to EITHER:
+        # - Allow for all of these methods to be called with `name`
+        # - OR, we need to expose a get test suite id by name method and save the `id`
+        return uuid4()
+
+    def run_external(self, external_evaluator: Callable[..., dict[str, ValidOutput]]) -> VellumTestSuiteResults:
```

Comment:
Can we pls call this something other than `external_evaluator` - when I read this I thought the evaluation was going to happen externally instead of the execution